// puregram/lib/common/attachments/document.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentAttachment = void 0;
const inspectable_1 = require("inspectable");
const structures_1 = require("../structures");
const file_attachment_1 = require("./file-attachment");
/**
 * This object represents a general file (as opposed to photos, voice messages
 * and audio files).
 */
// TODO: extended: ['fileId', 'fileUniqueId']
let DocumentAttachment = class DocumentAttachment extends file_attachment_1.FileAttachment {
    constructor() {
        super(...arguments);
        this.attachmentType = 'document';
    }
    /** Document thumbnail as defined by sender */
    get thumbnail() {
        const { thumbnail } = this.payload;
        if (!thumbnail) {
            return;
        }
        return new structures_1.PhotoSize(thumbnail);
    }
    /** Original filename as defined by sender */
    get fileName() {
        return this.payload.file_name;
    }
    /** MIME type of the file as defined by sender */
    get mimeType() {
        return this.payload.mime_type;
    }
    /** File size */
    get fileSize() {
        return this.payload.file_size;
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], DocumentAttachment.prototype, "thumbnail", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], DocumentAttachment.prototype, "fileName", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], DocumentAttachment.prototype, "mimeType", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], DocumentAttachment.prototype, "fileSize", null);
DocumentAttachment = __decorate([
    (0, inspectable_1.Inspectable)()
], DocumentAttachment);
exports.DocumentAttachment = DocumentAttachment;
