// puregram/lib/common/attachments/voice.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceAttachment = void 0;
const inspectable_1 = require("inspectable");
const file_attachment_1 = require("./file-attachment");
/** This object represents a voice note. */
// TODO: extended: ['fileId', 'fileUniqueId']
let VoiceAttachment = class VoiceAttachment extends file_attachment_1.FileAttachment {
    constructor() {
        super(...arguments);
        this.attachmentType = 'voice';
    }
    /** Duration of the audio in seconds as defined by sender */
    get duration() {
        return this.payload.duration;
    }
    /** MIME type of the file as defined by sender */
    get mimeType() {
        return this.payload.mime_type;
    }
    /** File size */
    get fileSize() {
        return this.payload.file_size;
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)()
], VoiceAttachment.prototype, "duration", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], VoiceAttachment.prototype, "mimeType", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], VoiceAttachment.prototype, "fileSize", null);
VoiceAttachment = __decorate([
    (0, inspectable_1.Inspectable)()
], VoiceAttachment);
exports.VoiceAttachment = VoiceAttachment;
