// puregram/lib/common/keyboards/inline-keyboard-builder.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var InlineKeyboardBuilder_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.InlineKeyboardBuilder = void 0;
const inspectable_1 = require("inspectable");
let InlineKeyboardBuilder = InlineKeyboardBuilder_1 = class InlineKeyboardBuilder {
    constructor() {
        // TODO: rows + currentRow
        this.rows = [];
        this.currentRow = [];
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /** Generate text button */
    textButton(params) {
        if (typeof params.payload === 'object') {
            params.payload = JSON.stringify(params.payload);
        }
        return this.addButton({
            text: params.text,
            callback_data: params.payload
        });
    }
    /** Generate URL button */
    urlButton(params) {
        if (typeof params.payload === 'object') {
            params.payload = JSON.stringify(params.payload);
        }
        return this.addButton({
            text: params.text,
            url: params.url,
            callback_data: params.payload || ''
        });
    }
    /** Generate Web App button */
    webAppButton(params) {
        return this.addButton({
            text: params.text,
            web_app: { url: params.url }
        });
    }
    /** Generate button that will switch to current chat and type the query */
    switchToCurrentChatButton(params) {
        return this.addButton({
            text: params.text,
            switch_inline_query_current_chat: params.query
        });
    }
    /** Generate button that will prompt user to select one of their chats */
    switchToChatButton(params) {
        return this.addButton({
            text: params.text,
            switch_inline_query: params.query
        });
    }
    /** Generate game button */
    gameButton(params) {
        return this.addWideButton({
            text: params.text,
            callback_game: params.game
        });
    }
    /** Generate pay button */
    payButton(params) {
        return this.addWideButton({
            pay: true,
            text: params.text
        });
    }
    /** Generate login button */
    loginButton(params) {
        return this.addButton({
            login_url: params.loginUrl,
            text: params.text
        });
    }
    /** Save current row of buttons in the general rows */
    row() {
        if (this.currentRow.length === 0) {
            return this;
        }
        this.rows.push(this.currentRow);
        this.currentRow = [];
        return this;
    }
    addButton(button) {
        this.currentRow.push(button);
        return this;
    }
    addWideButton(button) {
        if (this.currentRow.length !== 0) {
            this.row();
        }
        this.addButton(button);
        return this.row();
    }
    /** Clone current builder to new instance */
    clone() {
        const builder = new InlineKeyboardBuilder_1();
        builder.rows = [...this.rows];
        builder.currentRow = [...this.currentRow];
        return builder;
    }
    /** Returns JSON which is compatible with Telegram's `InlineKeyboardMarkup` interface */
    toJSON() {
        const buttons = this.currentRow.length !== 0
            ? [...this.rows, this.currentRow]
            : this.rows;
        return {
            inline_keyboard: buttons
        };
    }
    toString() {
        return JSON.stringify(this);
    }
};
__decorate([
    (0, inspectable_1.Inspect)({ as: 'inline_keyboard' })
], InlineKeyboardBuilder.prototype, "rows", void 0);
InlineKeyboardBuilder = InlineKeyboardBuilder_1 = __decorate([
    (0, inspectable_1.Inspectable)()
], InlineKeyboardBuilder);
exports.InlineKeyboardBuilder = InlineKeyboardBuilder;
