// puregram/lib/common/structures/chat-join-request.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatJoinRequest = void 0;
const inspectable_1 = require("inspectable");
const chat_1 = require("./chat");
const user_1 = require("./user");
const chat_invite_link_1 = require("./chat-invite-link");
/** Represents a join request sent to a chat. */
let ChatJoinRequest = class ChatJoinRequest {
    constructor(payload) {
        this.payload = payload;
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /** Chat to which the request was sent */
    get chat() {
        return new chat_1.Chat(this.payload.chat);
    }
    /** User that sent the join request */
    get from() {
        return new user_1.User(this.payload.from);
    }
    /** Identifier of a private chat with the user who sent the join request. This number may have more than 32 significant bits and some programming languages may have difficulty/silent defects in interpreting it. But it has at most 52 significant bits, so a 64-bit integer or double-precision float type are safe for storing this identifier. The bot can use this identifier for 24 hours to send messages until the join request is processed, assuming no other administrator contacted the user. */
    get userChatId() {
        return this.payload.user_chat_id;
    }
    /** Date the request was sent in Unix time */
    get date() {
        return this.payload.date;
    }
    /** Bio of the user */
    get bio() {
        return this.payload.bio;
    }
    /** Chat invite link that was used by the user to send the join request */
    get inviteLink() {
        const { invite_link } = this.payload;
        if (!invite_link) {
            return;
        }
        return new chat_invite_link_1.ChatInviteLink(invite_link);
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)()
], ChatJoinRequest.prototype, "chat", null);
__decorate([
    (0, inspectable_1.Inspect)()
], ChatJoinRequest.prototype, "from", null);
__decorate([
    (0, inspectable_1.Inspect)()
], ChatJoinRequest.prototype, "userChatId", null);
__decorate([
    (0, inspectable_1.Inspect)()
], ChatJoinRequest.prototype, "date", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], ChatJoinRequest.prototype, "bio", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], ChatJoinRequest.prototype, "inviteLink", null);
ChatJoinRequest = __decorate([
    (0, inspectable_1.Inspectable)()
], ChatJoinRequest);
exports.ChatJoinRequest = ChatJoinRequest;
