// puregram/lib/common/structures/chat-administrator-rights.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatAdministratorRights = void 0;
const inspectable_1 = require("inspectable");
/** Represents the rights of an administrator in a chat. */
let ChatAdministratorRights = class ChatAdministratorRights {
    constructor(payload) {
        this.payload = payload;
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /** `true`, if the user's presence in the chat is hidden */
    isAnonymous() {
        return this.payload.is_anonymous;
    }
    /** `true`, if the administrator can access the chat event log, chat statistics, message statistics in channels, see channel members, see anonymous administrators in supergroups and ignore slow mode. Implied by any other administrator privilege */
    canManageChat() {
        return this.payload.can_manage_chat;
    }
    /** `true`, if the administrator can delete messages of other users */
    canDeleteMessages() {
        return this.payload.can_delete_messages;
    }
    /** `true`, if the administrator can manage video chats */
    canManageVideoChats() {
        return this.payload.can_manage_video_chats;
    }
    /** `true`, if the administrator can restrict, ban or unban chat members */
    canRestrictMembers() {
        return this.payload.can_restrict_members;
    }
    /** `true`, if the administrator can add new administrators with a subset of their own privileges or demote administrators that he has promoted, directly or indirectly (promoted by administrators that were appointed by the user) */
    canPromoteMembers() {
        return this.payload.can_promote_members;
    }
    /** `true`, if the user is allowed to change the chat title, photo and other settings */
    canChangeInfo() {
        return this.payload.can_change_info;
    }
    /** `true`, if the user is allowed to invite new users to the chat */
    canInviteUsers() {
        return this.payload.can_invite_users;
    }
    /** `true`, if the administrator can post in the channel; channels only */
    canPostMessages() {
        return this.payload.can_post_messages;
    }
    /** `true`, if the administrator can edit messages of other users and can pin messages; channels only */
    canEditMessages() {
        return this.payload.can_edit_messages;
    }
    /** `true`, if the user is allowed to pin messages; groups and supergroups only */
    canPinMessages() {
        return this.payload.can_pin_messages;
    }
    /** `true`, if the user is allowed to create, rename, close, and reopen forum topics; supergroups only */
    canManageTopics() {
        return this.payload.can_manage_topics;
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "isAnonymous", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canManageChat", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canDeleteMessages", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canManageVideoChats", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canRestrictMembers", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canPromoteMembers", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canChangeInfo", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canInviteUsers", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canPostMessages", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canEditMessages", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canPinMessages", null);
__decorate([
    (0, inspectable_1.Inspect)({ compute: true })
], ChatAdministratorRights.prototype, "canManageTopics", null);
ChatAdministratorRights = __decorate([
    (0, inspectable_1.Inspectable)()
], ChatAdministratorRights);
exports.ChatAdministratorRights = ChatAdministratorRights;
