// puregram/lib/common/structures/chosen-inline-result.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChosenInlineResult = void 0;
const inspectable_1 = require("inspectable");
const user_1 = require("./user");
const location_1 = require("./location");
/** Represents a result of an inline query that was chosen by the user and sent to their chat partner. */
let ChosenInlineResult = class ChosenInlineResult {
    constructor(payload) {
        this.payload = payload;
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /** The unique identifier for the result that was chosen */
    get resultId() {
        return this.payload.result_id;
    }
    /** The user that chose the result */
    get from() {
        return new user_1.User(this.payload.from);
    }
    /** Sender ID */
    get senderId() {
        return this.from.id;
    }
    /** Sender location, only for bots that require user location */
    get location() {
        const { location } = this.payload;
        if (!location) {
            return;
        }
        return new location_1.Location(location);
    }
    /**
     * Identifier of the sent inline message. Available only if there is an
     * inline keyboard attached to the message. Will be also received in callback
     * queries and can be used to edit the message.
     */
    get inlineMessageId() {
        return this.payload.inline_message_id;
    }
    /** The query that was used to obtain the result */
    get query() {
        return this.payload.query;
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)()
], ChosenInlineResult.prototype, "resultId", null);
__decorate([
    (0, inspectable_1.Inspect)()
], ChosenInlineResult.prototype, "from", null);
__decorate([
    (0, inspectable_1.Inspect)()
], ChosenInlineResult.prototype, "senderId", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], ChosenInlineResult.prototype, "location", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], ChosenInlineResult.prototype, "inlineMessageId", null);
__decorate([
    (0, inspectable_1.Inspect)()
], ChosenInlineResult.prototype, "query", null);
ChosenInlineResult = __decorate([
    (0, inspectable_1.Inspectable)()
], ChosenInlineResult);
exports.ChosenInlineResult = ChosenInlineResult;
