// puregram/lib/common/structures/contact.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Contact = void 0;
const inspectable_1 = require("inspectable");
/** This object represents a phone contact. */
let Contact = class Contact {
    constructor(payload) {
        this.payload = payload;
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /** Contact's phone number */
    get phoneNumber() {
        return this.payload.phone_number;
    }
    /** Contact's first name */
    get firstName() {
        return this.payload.first_name;
    }
    /** Contact's last name */
    get lastName() {
        return this.payload.last_name;
    }
    /** Contact's user identifier in Telegram */
    get userId() {
        return this.payload.user_id;
    }
    /** Additional data about the contact in the form of a vCard */
    get vCard() {
        return this.payload.vcard;
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)()
], Contact.prototype, "phoneNumber", null);
__decorate([
    (0, inspectable_1.Inspect)()
], Contact.prototype, "firstName", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], Contact.prototype, "lastName", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], Contact.prototype, "userId", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], Contact.prototype, "vCard", null);
Contact = __decorate([
    (0, inspectable_1.Inspectable)()
], Contact);
exports.Contact = Contact;
