// puregram/lib/common/structures/file.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.File = void 0;
const inspectable_1 = require("inspectable");
/** This object represents a file ready to be downloaded. The file can be downloaded via the link `https://api.telegram.org/file/bot<token>/<file_path>`. It is guaranteed that the link will be valid for at least 1 hour. When the link expires, a new one can be requested by calling `getFile`. */
let File = class File {
    constructor(payload) {
        this.payload = payload;
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /**
     * Identifier for this file, which can be used to download or reuse the file
     */
    get fileId() {
        return this.payload.file_id;
    }
    /**
     * Unique identifier for this file, which is supposed to be the same over
     * time and for different bots. Can't be used to download or reuse the file.
     */
    get fileUniqueId() {
        return this.payload.file_unique_id;
    }
    /** File size, if known */
    get fileSize() {
        return this.payload.file_size;
    }
    /**
     * File path.
     * Use `https://api.telegram.org/file/bot<token>/<file_path>` to get the
     * file.
     */
    get filePath() {
        return this.payload.file_path;
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)()
], File.prototype, "fileId", null);
__decorate([
    (0, inspectable_1.Inspect)()
], File.prototype, "fileUniqueId", null);
__decorate([
    (0, inspectable_1.Inspect)()
], File.prototype, "fileSize", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], File.prototype, "filePath", null);
File = __decorate([
    (0, inspectable_1.Inspectable)()
], File);
exports.File = File;
