// puregram/lib/common/structures/photo-size.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhotoSize = void 0;
const inspectable_1 = require("inspectable");
/** This object represents one size of a photo or a file / sticker thumbnail */
let PhotoSize = class PhotoSize {
    constructor(payload) {
        this.payload = payload;
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /**
     * Identifier for this file, which can be used to download or reuse the file
     */
    get fileId() {
        return this.payload.file_id;
    }
    /**
     * Unique identifier for this file, which is supposed to be the same over
     * time and for different bots. Can't be used to download or reuse the file.
     */
    get fileUniqueId() {
        return this.payload.file_unique_id;
    }
    /** Photo width */
    get width() {
        return this.payload.width;
    }
    /** Photo height */
    get height() {
        return this.payload.height;
    }
    /** File size */
    get fileSize() {
        return this.payload.file_size;
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)()
], PhotoSize.prototype, "fileId", null);
__decorate([
    (0, inspectable_1.Inspect)()
], PhotoSize.prototype, "fileUniqueId", null);
__decorate([
    (0, inspectable_1.Inspect)()
], PhotoSize.prototype, "width", null);
__decorate([
    (0, inspectable_1.Inspect)()
], PhotoSize.prototype, "height", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], PhotoSize.prototype, "fileSize", null);
PhotoSize = __decorate([
    (0, inspectable_1.Inspectable)()
], PhotoSize);
exports.PhotoSize = PhotoSize;
