// puregram/lib/common/structures/venue.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Venue = void 0;
const inspectable_1 = require("inspectable");
const location_1 = require("./location");
/** This object represents a venue. */
let Venue = class Venue {
    constructor(payload) {
        this.payload = payload;
    }
    get [Symbol.toStringTag]() {
        return this.constructor.name;
    }
    /** Venue location */
    get location() {
        return new location_1.Location(this.payload.location);
    }
    /** Name of the venue */
    get title() {
        return this.payload.title;
    }
    /** Address of the venue */
    get address() {
        return this.payload.address;
    }
    /** Foursquare identifier of the venue */
    get foursquareId() {
        return this.payload.foursquare_id;
    }
    /** Foursquare type of the venue */
    get foursquareType() {
        return this.payload.foursquare_type;
    }
    /** Google Places identifier of the venue */
    get googlePlaceId() {
        return this.payload.google_place_id;
    }
    /**
     * Google Places type of the venue.
     * (See [supported types](https://developers.google.com/places/web-service/supported_types).)
     */
    get googlePlaceType() {
        return this.payload.google_place_type;
    }
    toJSON() {
        return this.payload;
    }
};
__decorate([
    (0, inspectable_1.Inspect)()
], Venue.prototype, "location", null);
__decorate([
    (0, inspectable_1.Inspect)()
], Venue.prototype, "title", null);
__decorate([
    (0, inspectable_1.Inspect)()
], Venue.prototype, "address", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], Venue.prototype, "foursquareId", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], Venue.prototype, "foursquareType", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], Venue.prototype, "googlePlaceId", null);
__decorate([
    (0, inspectable_1.Inspect)({ nullable: false })
], Venue.prototype, "googlePlaceType", null);
Venue = __decorate([
    (0, inspectable_1.Inspectable)()
], Venue);
exports.Venue = Venue;
